spec/unit/util/execution_spec.rb

```diff
@@ -550,39 +521,19 @@ def stub_process_wait(exitstatus)
       end
 
       it "should read and return the output if squelch is false" do
-        stdout = Puppet::FileSystem::Uniquefile.new('test')
-        Puppet::FileSystem::Uniquefile.stubs(:new).returns(stdout)
-        stdout.write("My expected command output")
+        r, w = IO.pipe
+        IO.expects(:pipe).returns([r, w])
+        w.write("My expected command output")
 
         expect(Puppet::Util::Execution.execute('test command')).to eq("My expected command output")
       end
 
       it "should not read the output if squelch is true" do
-        stdout = Puppet::FileSystem::Uniquefile.new('test')
-        Puppet::FileSystem::Uniquefile.stubs(:new).returns(stdout)
-        stdout.write("My expected command output")
+        IO.expects(:pipe).never
 
         expect(Puppet::Util::Execution.execute('test command', :squelch => true)).to eq('')
       end
 
-      it "should delete the file used for output if squelch is false" do
-        stdout = Puppet::FileSystem::Uniquefile.new('test')
-        path = stdout.path
-        Puppet::FileSystem::Uniquefile.stubs(:new).returns(stdout)
-
-        Puppet::Util::Execution.execute('test command')
-
-        expect(Puppet::FileSystem.exist?(path)).to be_falsey
-      end
```

Comment:
should we add a test to make sure both ends of the pipe are closed (during normal run and if an exception is raised)?